Demo/Application/Features/VenmoViewController.swift

```diff
@@ -67,6 +66,8 @@ class VenmoViewController: PaymentButtonBaseViewController {
     func checkout(request: BTVenmoRequest)  {
         Task {
             do {
+                let apiClient = BTAPIClient(authorization: "sandbox_9dbg82cq_dcpspy2brwdjr3qn")!
+                let venmoClient = BTVenmoClient(apiClient: apiClient)
```

Comment:
While this helps us test this merchant integration style, it now renders our Settings screen obsolete since the env & auth type settings will be ignored for every feature VC.

We could include these changes in a git patch and apply them locally for our own testing? This PR can be just the BTAnalyticsService changes - what do you think?

Reply:
Oh yeah, these were not meant to be permanent changes. It was just to reproduce the conditions

Reply:
Do you want me to write a test to fail before the BTAnalyticsService change on this PR?
BTAnalyticsService's omission of that config call is just one of 3 that I usually see with this merchant integration scenario.

Reply:
Let me look into git patch 😅
And as to writing unit test failing on main, I don't see a way of simulating quick fetchOrRemoteConfig calls involving ConfigCache right now. Maybe with ConfigutationLoader, we can test this scenario.

Reply:
> And as to writing unit test failing on main, I don't see a way of simulating quick succession of fetchOrRemoteConfig calls involving ConfigCache right now. 

That's OK. This has been a tricky problem to replicate. I think it's OK if we can't write the perfect failing test and we can still make small changes that we think will help this problem.

Reply:
I wrote the test. 45eface812323ba501935cf61dac303722ce87f3

Reply:
I'm concerned about concurrent cache writes that can happen with current implementation for some merchant integrations that might trigger quick successive calls to fetchOrReturnRemoteConfiguration.